src/Compilers/Core/Portable/CodeGen/ITokenDeferral.cs

```diff
@@ -12,13 +12,14 @@ internal interface ITokenDeferral
     {
         uint GetFakeStringTokenForIL(string value);
         uint GetFakeSymbolTokenForIL(Cci.IReference value, SyntaxNode syntaxNode, DiagnosticBag diagnostics);
+        uint GetFakeSymbolTokenForIL(Cci.ISignature value, SyntaxNode syntaxNode, DiagnosticBag diagnostics);
         uint GetSourceDocumentIndexForIL(Cci.DebugSourceDocument document);
 
         Cci.IFieldReference GetFieldForData(ImmutableArray<byte> data, SyntaxNode syntaxNode, DiagnosticBag diagnostics);
         Cci.IMethodReference GetInitArrayHelper();
 
         string GetStringFromToken(uint token);
-        Cci.IReference GetReferenceFromToken(uint token);
+        object GetReferenceFromToken(uint token);
```

Comment:
>GetReferenceFromToken [](start = 15, length = 21)

Consider adding a doc comment listing the types that can come out of this method.